tests/functional/test_configured_endpoint_url.py

```diff
@@ -0,0 +1,283 @@
+        result.add((model.service_name, model))
+    return sorted(result)
+
+
+def create_service_env_test_case(service_name, service_model):
+    transformed_service_id = service_model.service_id.replace(' ', '_').upper()
+
+    return {
+        'service': service_name,
+        'profile': 'default',
+        'augmented_data': {
+            'profile': {
+                'aws_access_key_id': '123',
+                'aws_secret_access_key': '456',
+                'region': 'fake-region-10',
+            },
+            'services': {},
+            'environment': {
+                f'AWS_ENDPOINT_URL_{transformed_service_id}': 'https://endpoint-override'
+            },
+            'client_config': {},
+        },
+    }
+
+
+def create_service_config_test_case(service_name, service_model):
+    transformed_service_id = service_model.service_id.replace(' ', '_').lower()
+
+    return {
+        'service': service_name,
+        'profile': 'default',
+        'augmented_data': {
+            'profile': {
+                'services': 'my-services',
+                'aws_access_key_id': '123',
+                'aws_secret_access_key': '456',
+                'region': 'fake-region-10',
+            },
+            'services': {
+                transformed_service_id: {
+                    'endpoint_url': 'https://endpoint-override'
+                }
+            },
+            'environment': {},
+            'client_config': {},
+        },
+    }
+
+
+SERVICE_TO_OPERATION = {
+    'acm-pca': 'list_certificate_authorities',
+    'apigateway': 'get_account',
+    'cloudfront': 'list_distributions',
+    'cloudtrail': 'list_trails',
+    'codebuild': 'list_projects',
+    # 'ec2': 'describe_instances',
+    's3': 'list_buckets',
+    # 'sts': 'get_caller_identity',
+}
+
+
+def create_service_to_operation_cases():
+    available_services = [
+        x
+        for x in _known_service_names_and_ids()
+        if x[0] in SERVICE_TO_OPERATION.keys()
+    ]
+
+    for x in available_services:
+        yield (x[0], x[1], SERVICE_TO_OPERATION[x[0]])
+
+
+@pytest.fixture
+def botocore_session():
+    return botocore.session.get_session()
+
+
+@pytest.mark.parametrize(
+    "test_case", create_cases(ENDPOINT_TEST_SUITE), ids=idfn
+)
+def test_resolve_configured_endpoint_url(test_case, botocore_session):
+    setup_configuration_environment(test_case)
+    client = get_env_config_file_patched_client(botocore_session, test_case)
+    assert_endpoint_url_used(
+        client=client,
+        expected_endpoint_url=test_case['output']['endpointUrl'],
+    )
+
+
+@pytest.mark.parametrize(
+    'service_name,service_model', _known_service_names_and_ids()
+)
+def test_service_env_var_name_is_correct(
+    service_name, service_model, botocore_session
+):
+    test_case = create_service_env_test_case(service_name, service_model)
+    setup_configuration_environment(test_case)
+
+    client = get_env_config_file_patched_client(botocore_session, test_case)
```

Comment:
It would be interesting to see if we can simplify these test cases down. Mainly it seems the complexity is in we are trying to reuse the test case schema in order to reuse the scaffolding and I think if we can move away from building these test cases using the test schema it can help simplifying the tests. I think with some of the suggestions that I had early should help with this. So for example for this test it would be:
```python
@pytest.mark.parametrize(
    'service_name,service_model', _known_service_names_and_ids()
)
def test_service_env_var_name_is_respected(
    service_name, service_model, create_client
):
    expected_endpoint_url_env_var = (
        ''AWS_ENDPOINT_URL_{service_model.service_id.replace(' ', '_').upper()}'
    )
    client = create_client(
        service_name,
        environment={
           expected_endpoint_url_env_var: 'https://endpoint-override'
        }
    assert_endpoint_url_used(
        client=client, expected_endpoint_url='https://endpoint-override'
    )
```

Reply:
I'm pretty sure I still need to provide a profile here (even a default one).